app/controllers/patients_controller.rb

```diff
@@ -116,15 +122,32 @@ def edit
 
   def update
     @patient.last_edited_by = current_user
-    if @patient.update patient_params
-      @patient = Patient.includes(versions: [:item, :user])
-                        .find(@patient.id) # reload
-      flash.now[:notice] = t('flash.patient_info_saved', timestamp: Time.zone.now.display_timestamp)
-    else
-      error = @patient.errors.full_messages.to_sentence
-      flash.now[:alert] = error
+
+    respond_to do |format|
+      format.js do
+        if @patient.update patient_params
+          @patient = Patient.includes(versions: [:item, :user]).find(@patient.id) # reload
+          flash.now[:notice] = t('flash.patient_info_saved', timestamp: Time.zone.now.display_timestamp)
+        else
+          error = @patient.errors.full_messages.to_sentence
+          flash.now[:alert] = error
+        end
+      end
+
+      format.json do
+        if @patient.update patient_params
+          @patient.reload
+          render json: {
+            patient: @patient.reload.as_json,
+            flash: {
+              notice: t('flash.patient_info_saved', timestamp: Time.zone.now.display_timestamp)
+            }
+          }, status: :ok
+        else
+          render json: { flash: { alert: @patient.errors.full_messages.to_sentence } }, status: :unprocessable_entity
+        end
+      end
```

Comment:
As someone who constantly mixes up my js/json contexts, a comment above each `format` on when the app asks for each would be great!